File: 06/a.js
const fs = require('fs')
const input = fs.readFileSync(__dirname + '/input.txt', 'utf8')

const [ time, distance ] = input
  .split('\n')
  .filter(ln => ln.length)
  .map(ln => ln
    .trim()
    .split(/\s+/)
    .map(str => isNaN(Number(str)) ? str : Number(str))
  )

let product = 1

for (let i = 1; i < time.length; i++) {
  const ms = time[i]
  const record = distance[i]
  let count = 0

  for (let hold = 1; hold < ms; hold ++) {
    const traveled = hold * (ms - hold)
    if (traveled > record) {
      count ++
    } else if (count > 0) {
      break
    }
  }
  product *= count
}

console.log(product)

File: 06/a.py
import pathlib
import re
time, distance = [[int(x) if x.isdigit()
  else x for x in re.split(r'\s+', ln.strip())] for ln in open(
    str(pathlib.Path(__file__).parent.resolve()) + '/input.txt'
  ).readlines()]

product = 1

for i in range(1, len(time)):
  ms = time[i]
  record = distance[i]
  count = 0

  for hold in range(1, ms):
    traveled = hold * (ms - hold)
    if traveled > record:
      count += 1
    elif count > 0:
      break
  product *= count

print(product)